libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -4747,6 +4751,35 @@ void GCS_MAVLINK::send_set_position_target_global_int(uint8_t target_system, uin
             0,0);   // yaw, yaw_rate
 }
 
+// returns true if all battery instances were reported
+bool GCS_MAVLINK::send_smart_battery_info()
+{
+#if HAL_SMART_BATTERY_INFO_ENABLED
+    const AP_BattMonitor &battery = AP::battery();
+
+    for (uint8_t i = 0; i < battery.num_instances(); i++) {
+        const uint8_t battery_id = (last_smart_battery_info_idx + 1) % battery.num_instances();
+        AP_BattMonitor::Type batt_type = battery.get_type(battery_id);
+        if (batt_type != AP_BattMonitor::Type::NONE &&
+            batt_type != AP_BattMonitor::Type::ANALOG_VOLTAGE_ONLY &&
+            batt_type != AP_BattMonitor::Type::ANALOG_VOLTAGE_AND_CURRENT) {
+
+            CHECK_PAYLOAD_SIZE(SMART_BATTERY_INFO);
+            if (!AP::battery().send_mavlink_smart_battery_info(battery_id, chan)) {
+                return false;
+            }
+            last_smart_battery_info_idx = battery_id;
+        } else {
+            last_smart_battery_info_idx = battery_id;
+        }
+    }
+    return true;
+#else
+    return false;
```

Comment:
This must return true.  These send messages can only return false in the case there's no buffer space, essentially.

You also need to test this PR with SMART_BATTERY_INFO disabled.